apps/omg_watcher_rpc/test/omg_watcher_rpc/web/controllers/transaction_test.exs

```diff
@@ -1800,6 +1801,275 @@ defmodule OMG.WatcherRPC.Web.Controller.TransactionTest do
     end
   end
 
+  describe "/transaction.merge" do
+    @tag fixtures: [:phoenix_ecto_sandbox, :alice]
+    test "address currency params return expected result", %{alice: alice} do
+      alice_hex = Encoding.to_hex(alice.addr)
+
+      insert(:txoutput)
+      _ = insert(:txoutput, amount: 1, currency: @eth, owner: alice.addr)
+      _ = insert(:txoutput, amount: 1, currency: @eth, owner: alice.addr)
+      _ = insert(:txoutput, amount: 1, currency: @eth, owner: alice.addr)
+      _ = insert(:txoutput, amount: 1, currency: @other_token, owner: alice.addr)
+      _ = insert(:txoutput, amount: 1, currency: @other_token, owner: alice.addr)
+      _ = insert(:txoutput, amount: 1, currency: @other_token, owner: alice.addr)
+
+      assert %{
+               "transactions" => [
+                 %{
+                   "typed_data" => _,
+                   "txbytes" => _,
+                   "inputs" => [
+                     %{
+                       "amount" => 1,
+                       "currency" => @eth_hex,
+                       "owner" => ^alice_hex
+                     },
+                     %{
+                       "amount" => 1,
+                       "currency" => @eth_hex,
+                       "owner" => ^alice_hex
+                     },
+                     %{
+                       "amount" => 1,
+                       "currency" => @eth_hex,
+                       "owner" => ^alice_hex
+                     }
+                   ],
+                   "outputs" => [
+                     %{
+                       "amount" => 3,
+                       "currency" => @eth_hex,
+                       "owner" => ^alice_hex
+                     }
+                   ]
+                 }
+               ]
+             } =
+               WatcherHelper.success?(
+                 "transaction.merge",
+                 %{
+                   "address" => alice_hex,
+                   "currency" => @eth_hex
+                 }
+               )
+    end
+
+    @tag fixtures: [:phoenix_ecto_sandbox, :alice]
+    test "address currency params returns multiple valid transactions", %{alice: alice} do
+      alice_hex = Encoding.to_hex(alice.addr)
+
+      insert(:txoutput)
+      _ = insert(:txoutput, amount: 1, currency: @eth, owner: alice.addr)
```

Comment:
You could move all these inserts in a `setup` and re-use them in different tests

Reply:
I would rather keep the `_ = insert` clauses in the individual tests so that a reader can more easily trace what scenario is being set up for the test. 

Trying to use `setup` for the initial `insert(:txoutput` that is needed - but need to figure out how to start the `watcher_info` database.